pool/sqrt_price_math.gno

```diff
@@ -106,37 +181,69 @@ func sqrtPriceMathGetNextSqrtPriceFromOutput(
+		sqrtPX96,
+		ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromOutput() || expected sqrtPX96 >= 0. got: %s",
+		sqrtPX96,
+	))
+
+	requireUnsigned(
+		liquidity,
+		ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromOutput() || expected liquidity >= 0. got: %d",
+		liquidity,
+	))
 
 	if zeroForOne {
-		requireUnsigned(sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown(sqrtPX96, liquidity, amountOut, false), "[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromOutput() || condition must return `v` >= 0__#1")
-		return sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown(sqrtPX96, liquidity, amountOut, false)
+		amount1Result := sqrtPriceMathGetNextSqrtPriceFromAmount1RoundingDown(sqrtPX96, liquidity, amountOut, false)
+		requireUnsigned(
+			amount1Result,
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromOutput() || condition must return `v` >= 0__#1",
+		)
+
+		return amount1Result
 	}
 
-	requireUnsigned(sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp(sqrtPX96, liquidity, amountOut, false), "[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromOutput() || condition must return `v` >= 0__#2")
-	return sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp(sqrtPX96, liquidity, amountOut, false)
+	amount0Result := sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp(sqrtPX96, liquidity, amountOut, false)
+	requireUnsigned(
+		amount0Result,
+		"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromOutput() || condition must return `v` >= 0__#2",
+	)
+
+	return amount0Result
 }
 
 func sqrtPriceMathGetAmount0DeltaHelper(
 	sqrtRatioAX96 bigint,
 	sqrtRatioBX96 bigint,
 	liquidity bigint,
 ) bigint {
-	requireUnsigned(sqrtRatioAX96, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetAmount0DeltaHelper() || sqrtRatioAX96(%d) >= 0", sqrtRatioAX96))
-	requireUnsigned(sqrtRatioBX96, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetAmount0DeltaHelper() || sqrtRatioBX96(%d) >= 0", sqrtRatioBX96))
-	requireUnsigned(liquidity, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetAmount0DeltaHelper() || liquidity(%d) >= 0", liquidity))
+	requireUnsigned(
+		sqrtRatioAX96,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetAmount0DeltaHelper() || expected sqrtRatioAX96 >= 0. got: %d",
+			sqrtRatioAX96,
+	))
+
+	requireUnsigned(
+		sqrtRatioBX96,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetAmount0DeltaHelper() || expected sqrtRatioBX96 >= 0. got: %d",
+			sqrtRatioBX96,
+	))
+
+	requireUnsigned(
+		liquidity,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetAmount0DeltaHelper() || expected liquidity >= 0. got: %d",
+			liquidity,
+	))
 
 	if sqrtRatioAX96 > sqrtRatioBX96 {
 		sqrtRatioAX96, sqrtRatioBX96 = sqrtRatioBX96, sqrtRatioAX96
 	}
-	requireUnsigned(sqrtRatioAX96, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetAmount0DeltaHelper() || sqrtRatioAX96(%d) >= 0", sqrtRatioAX96))
```

Comment:
can we guarantee three values are Unsigned without checking it?

Reply:
Yes, all three values are unsigned.

1. initially verified that each values are unsigned.
2. if `AX96` is greater than `BX96`, the two values are interchanged, so the calculation of `numerator2` will also be unsigned.
3. also perform a bitwise operation (`<<`) on `numerator1` by 96, it will not become negative because it is a multiplication operation.